File: bank-service/src/main/java/rs/edu/raf/service/CountryService.java
package rs.edu.raf.service;

import rs.edu.raf.dto.CountryCreateDTO;
import rs.edu.raf.dto.CountryDTO;

import java.util.List;

public interface CountryService {
    /**
     * Creates country
     * @param countryCreateDTO DTO which contains information about country
     * @return {@link CountryDTO} object representing created country
     */
    CountryDTO createCountry(CountryCreateDTO countryCreateDTO);

    /**
     * Gets all countries
     * @return List of {@link CountryDTO} representing countries
     */
    List<CountryDTO> getAllCountries();

    /**
     * Deletes country
     * @param id the id of the country to be deleted
     */
    void deleteCountry(Long id);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/CountryService.kt
package org.example.bankservicekotlin.service

import org.example.bankservicekotlin.dto.CountryCreateDTO
import org.example.bankservicekotlin.dto.CountryDTO

interface CountryService {
    /**
     * Creates country
     * @param countryCreateDTO DTO which contains information about country
     * @return [CountryDTO] object representing created country
     */
    fun createCountry(countryCreateDTO: CountryCreateDTO): CountryDTO

    /**
     * Gets all countries
     * @return List of [CountryDTO] representing countries
     */
    fun getAllCountries(): List<CountryDTO>

    /**
     * Deletes country
     * @param id the id of the country to be deleted
     */
    fun deleteCountry(id: Long)
}